include/picongpu/simulation/stage/ParticleBoundaries.hpp

```diff
@@ -98,54 +98,77 @@ namespace picongpu
-                            }
-                            if(kindName == "absorbing")
-                            {
-                                // For now it must match the default-set boundary kind
-                                if(T_Species::boundaryDescription()[d].kind != particles::boundary::Kind::Absorbing)
+                                        "Negative boundary temperature " + errorString + " is not supported");
+                                T_Species::boundaryDescription()[d].temperature = temperature;
+
+                                auto const kindName = kindNames()[d];
+                                if(kindName == "reflecting")
+                                {
+                                    if(T_Species::boundaryDescription()[d].kind == particles::boundary::Kind::Periodic)
+                                        throw std::runtime_error(
+                                            "Boundary kind " + errorString
+                                            + " is not compatible with --periodic value");
+                                    T_Species::boundaryDescription()[d].kind = particles::boundary::Kind::Reflecting;
+                                }
+                                if(kindName == "thermal")
+                                {
+                                    if(T_Species::boundaryDescription()[d].kind == particles::boundary::Kind::Periodic)
+                                        throw std::runtime_error(
+                                            "Boundary kind " + errorString
+                                            + " is not compatible with --thermal value");
+                                    T_Species::boundaryDescription()[d].kind = particles::boundary::Kind::Thermal;
+                                }
+                                if(kindName == "periodic")
+                                {
+                                    // For now it must match the default-set boundary kind
+                                    if(T_Species::boundaryDescription()[d].kind != particles::boundary::Kind::Periodic)
+                                        throw std::runtime_error(
+                                            "Boundary kind " + errorString
+                                            + " is not compatible with --periodic value");
+                                }
+                                if(kindName == "absorbing")
+                                {
+                                    // For now it must match the default-set boundary kind
+                                    if(T_Species::boundaryDescription()[d].kind
+                                       != particles::boundary::Kind::Absorbing)
+                                        throw std::runtime_error(
+                                            "Boundary kind " + errorString
+                                            + " is not compatible with --periodic value");
```

Comment:
Again, I believe either here or in the check above something is wrong. Probably the error message.
Isn't it necessary to set the `T_Species::boundaryDescription()[d].kind` here or in the check above as well?

Reply:
What should be wrong here? I have not changed anything in this PR.

Reply:
If there is an error, we need a separate PR as suggested by @psychocoderHPC